### ISSUE TITLE
`dagger run` does not work with Gradle

### ISSUE THREAD

**Issue by unredundant** [dagger/dagger#5287]
### What is the issue?

[Gradle](https://gradle.org/) does not work when run within a `dagger run` context.  

### Log output

Example.  I have a simple application in a gradle module called playground. I can run the app easily with the following 

```
$ ./gradlew playground:run -q --console=plain
hi
```

however, feed this through `dagger run ./gradlew playground:run -q --console=plain` and the process hangs indefinitely.  

Adding a `--debug` does not offer a whole lot of insight 

```
$  dagger run -s ./gradlew playground:run --debug --console=plain
Connected to engine b406cfb838a0
2023-06-04T14:00:30.865-0400 [INFO] [org.gradle.internal.nativeintegration.services.NativeServices] Initialized native services in: /Users/megame/.gradle/native
2023-06-04T14:00:30.890-0400 [INFO] [org.gradle.internal.nativeintegration.services.NativeServices] Initialized jansi services in: /Users/megame/.gradle/native
```

At this point, it just hangs indefinitely.

### Steps to reproduce

This can be reproduced even without an associated project.  Install Gradle (my version is 8.1.1) and run `gradle tasks`.  Assuming you are in a directory with no gradle project, this operation will fail, but it will show some output.  Do the same via `dagger run -s gradle tasks --console=plain` and you will see that even this command hangs. 

### SDK version

Dagger CLI 0.6.1

### OS version

Confirmed on both MacOS Mojave and Ubuntu 22.04

**Comment by helderco** [comment]
My guess is that gradle needs some control over stdout/stderr/stdin \cc @vito.

I reproduced it, even with an error (i.e., on it's own the usage error from gradle reports I'm missing some tooling, so it fails early).

**Comment by wingyplus** [comment]
A bit more info from `./gradlew --debug`:

```
$ dagger run ./gradlew build --debug --console=plain
█ ERROR ./gradlew build --debug --console=plain
┃ 2023-06-06T19:43:11.769+0700 [INFO] [org.gradle.internal.nativeintegration.services.NativeServices] Initialized native services in: /
┃ Users/thanabodee/.gradle/native
┃ 2023-06-06T19:43:11.924+0700 [INFO] [org.gradle.internal.nativeintegration.services.NativeServices] Initialized jansi services in: /U
┃ sers/thanabodee/.gradle/native
┃ 2023-06-06T19:43:11.945+0700 [LIFECYCLE] [org.gradle.launcher.cli.DebugLoggerWarningAction]
┃ #############################################################################
┃    WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING
┃
┃    Debug level logging will leak security sensitive information!
┃
┃    https://docs.gradle.org/8.0.1/userguide/logging.html#sec:debug_security
┃ #############################################################################
┃
┃ 2023-06-06T19:43:12.030+0700 [INFO] [org.gradle.internal.jvm.inspection.DefaultJvmMetadataDetector] Received JVM installation metadat
┃ a from '/Users/thanabodee/Library/Java/JavaVirtualMachines/openjdk-18/Contents/Home': {JAVA_HOME=/Users/thanabodee/Library/Java/JavaV
┃ irtualMachines/openjdk-18/Contents/Home, JAVA_VERSION=18, JAVA_VENDOR=Oracle Corporation, RUNTIME_NAME=OpenJDK Runtime Environment, R
┃ UNTIME_VERSION=18+36-2087, VM_NAME=OpenJDK 64-Bit Server VM, VM_VERSION=18+36-2087, VM_VENDOR=Oracle Corporation, OS_ARCH=x86_64}
┃ 2023-06-06T19:43:12.190+0700 [LIFECYCLE] [org.gradle.launcher.daemon.client.SingleUseDaemonClient] To honour the JVM settings for thi
┃ s build a single-use Daemon process will be forked. See https://docs.gradle.org/8.0.1/userguide/gradle_daemon.html#sec:disabling_the_
┃ daemon.
┃ 2023-06-06T19:43:12.281+0700 [DEBUG] [org.gradle.launcher.daemon.client.DefaultDaemonStarter] Using daemon args: [/Users/thanabodee/L
┃ ibrary/Java/JavaVirtualMachines/openjdk-18/Contents/Home/bin/java, --add-opens=java.base/java.util=ALL-UNNAMED, --add-opens=java.base
┃ /java.lang=ALL-UNNAMED, --add-opens=java.base/java.lang.invoke=ALL-UNNAMED, --add-opens=java.prefs/java.util.prefs=ALL-UNNAMED, --add
┃ -opens=java.base/java.nio.charset=ALL-UNNAMED, --add-opens=java.base/java.net=ALL-UNNAMED, --add-opens=java.base/java.util.concurrent
┃ .atomic=ALL-UNNAMED, -XX:MaxMetaspaceSize=256m, -XX:+HeapDumpOnOutOfMemoryError, -Xms256m, -Xmx512m, -Dfile.encoding=UTF-8, -Duser.co
┃ untry=TH, -Duser.language=en, -Duser.variant, -cp, /Users/thanabodee/.gradle/wrapper/dists/gradle-8.0.1-bin/7rsvvmdp4bsd5fhm7a9cyrnzz
┃ /gradle-8.0.1/lib/gradle-launcher-8.0.1.jar]
┃ 2023-06-06T19:43:12.287+0700 [DEBUG] [org.gradle.launcher.daemon.client.DefaultDaemonStarter] Starting daemon process: workingDir = /
┃ Users/thanabodee/.gradle/daemon/8.0.1, daemonArgs: [/Users/thanabodee/Library/Java/JavaVirtualMachines/openjdk-18/Contents/Home/bin/j
┃ ava, --add-opens=java.base/java.util=ALL-UNNAMED, --add-opens=java.base/java.lang=ALL-UNNAMED, --add-opens=java.base/java.lang.invoke
┃ =ALL-UNNAMED, --add-opens=java.prefs/java.util.prefs=ALL-UNNAMED, --add-opens=java.base/java.nio.charset=ALL-UNNAMED, --add-opens=jav
┃ a.base/java.net=ALL-UNNAMED, --add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED, -XX:MaxMetaspaceSize=256m, -XX:+HeapDumpO
┃ nOutOfMemoryError, -Xms256m, -Xmx512m, -Dfile.encoding=UTF-8, -Duser.country=TH, -Duser.language=en, -Duser.variant, -cp, /Users/than
┃ abodee/.gradle/wrapper/dists/gradle-8.0.1-bin/7rsvvmdp4bsd5fhm7a9cyrnzz/gradle-8.0.1/lib/gradle-launcher-8.0.1.jar, org.gradle.launch
┃ er.daemon.bootstrap.GradleDaemon, 8.0.1]
┃ 2023-06-06T19:43:12.306+0700 [INFO] [org.gradle.process.internal.DefaultExecHandle] Starting process 'Gradle build daemon'. Working d
┃ irectory: /Users/thanabodee/.gradle/daemon/8.0.1 Command: /Users/thanabodee/Library/Java/JavaVirtualMachines/openjdk-18/Contents/Home
┃ /bin/java --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.lang.inv
┃ oke=ALL-UNNAMED --add-opens=java.prefs/java.util.prefs=ALL-UNNAMED --add-opens=java.base/java.nio.charset=ALL-UNNAMED --add-opens=jav
┃ a.base/java.net=ALL-UNNAMED --add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOu
┃ tOfMemoryError -Xms256m -Xmx512m -Dfile.encoding=UTF-8 -Duser.country=TH -Duser.language=en -Duser.variant -cp /Users/thanabodee/.gra
┃ dle/wrapper/dists/gradle-8.0.1-bin/7rsvvmdp4bsd5fhm7a9cyrnzz/gradle-8.0.1/lib/gradle-launcher-8.0.1.jar org.gradle.launcher.daemon.bo
┃ otstrap.GradleDaemon 8.0.1
┃ 2023-06-06T19:43:12.307+0700 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Changing state to: STARTING
┃ 2023-06-06T19:43:12.310+0700 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Waiting until process started: Gradle build daem
┃ on.
┃ 2023-06-06T19:43:12.335+0700 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Changing state to: STARTED
┃ 2023-06-06T19:43:12.335+0700 [DEBUG] [org.gradle.process.internal.ExecHandleRunner] waiting until streams are handled...
┃ 2023-06-06T19:43:12.336+0700 [INFO] [org.gradle.process.internal.DefaultExecHandle] Successfully started process 'Gradle build daemon
┃ '
┃ 2023-06-06T19:43:12.336+0700 [DEBUG] [org.gradle.launcher.daemon.client.DefaultDaemonStarter] Gradle daemon process is starting. Wait
┃ ing for the daemon to detach...
┃ 2023-06-06T19:43:12.336+0700 [DEBUG] [org.gradle.launcher.daemon.bootstrap.DaemonOutputConsumer] Starting consuming the daemon proces
┃ s output.
┃ 2023-06-06T19:43:14.085+0700 [DEBUG] [org.gradle.launcher.daemon.bootstrap.DaemonOutputConsumer] daemon out: Daemon started. About to
┃  close the streams. Daemon details: 0100053232333839002466646566333430302d653363612d343733652d626662652d6663613465333035323430655ad06
┃ 6b6ed424785bc84b0f5f7152b1d0000ed6600000001000000047f000001003b2f55736572732f7468616e61626f6465652f2e677261646c652f6461656d6f6e2f382e
┃ 302e312f6461656d6f6e2d32323338392e6f75742e6c6f67
┃ 2023-06-06T19:43:14.088+0700 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Changing state to: DETACHED
┃ 2023-06-06T19:43:14.088+0700 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Process 'Gradle build daemon' finished with exit
┃  value 0 (state: DETACHED)
┃ 2023-06-06T19:43:14.088+0700 [DEBUG] [org.gradle.launcher.daemon.client.DefaultDaemonStarter] Gradle daemon process is now detached.
┃ 2023-06-06T19:43:14.100+0700 [INFO] [org.gradle.launcher.daemon.client.DefaultDaemonStarter] An attempt to start the daemon took 1.81
┃ 2 secs.
┃ 2023-06-06T19:43:14.100+0700 [DEBUG] [org.gradle.launcher.daemon.client.DefaultDaemonConnector] Started Gradle daemon DaemonStartupIn
┃ fo{pid=22389, uid=fdef3400-e3ca-473e-bfbe-fca4e305240e, address=[5ad066b6-ed42-4785-bc84-b0f5f7152b1d port:60774, addresses:[/127.0.0
┃ .1]], diagnostics={pid=22389, daemonLog=/Users/thanabodee/.gradle/daemon/8.0.1/daemon-22389.out.log}}
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interfa
┃ ce utun6
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? false
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding remote address /fe80:0:0:0:267:3
┃ a06:12d6:5157%utun6
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interfa
┃ ce utun5
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? false
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding remote address /fe80:0:0:0:615e:
┃ d3d:9a6d:450c%utun5
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interfa
┃ ce utun4
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? false
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding remote address /fe80:0:0:0:1663:
┃ ba57:1915:9099%utun4
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interfa
┃ ce utun3
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? false
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding remote address /fe80:0:0:0:f1c3:
┃ 70c:2a23:962f%utun3
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interfa
┃ ce utun2
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? false
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding remote address /fe80:0:0:0:ce81:
┃ b1c:bd2c:69e%utun2
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interfa
┃ ce utun1
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? false
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding remote address /fe80:0:0:0:67d8:
┃ 91e8:f094:4868%utun1
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interfa
┃ ce utun0
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? false
┃ 2023-06-06T19:43:14.108+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding remote address /fe80:0:0:0:250f:
┃ 2b49:d74b:fe6d%utun0
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interfa
┃ ce llw0
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? false
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding remote address /fe80:0:0:0:5094:
┃ b3ff:fea2:9309%llw0
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interfa
┃ ce awdl0
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? false
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding remote address /fe80:0:0:0:5094:
┃ b3ff:fea2:9309%awdl0
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interfa
┃ ce ap1
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? false
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding remote address /fe80:0:0:0:bcd0:
┃ 74ff:fe36:bc8b%ap1
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interfa
┃ ce anpi1
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? false
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding remote address /fe80:0:0:0:85e:8
┃ aff:fe24:355b%anpi1
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interfa
┃ ce anpi2
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? false
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding remote address /fe80:0:0:0:85e:8
┃ aff:fe24:355c%anpi2
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interfa
┃ ce anpi0
┃ 2023-06-06T19:43:14.109+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? false
┃ 2023-06-06T19:43:14.110+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding remote address /fe80:0:0:0:85e:8
┃ aff:fe24:355a%anpi0
┃ 2023-06-06T19:43:14.110+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interfa
┃ ce en0
┃ 2023-06-06T19:43:14.110+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? false
┃ 2023-06-06T19:43:14.110+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding remote address /fe80:0:0:0:1056:
┃ 8dee:6e45:bd6b%en0
┃ 2023-06-06T19:43:14.110+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding remote address /10.40.145.96
┃ 2023-06-06T19:43:14.110+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding IP addresses for network interfa
┃ ce lo0
┃ 2023-06-06T19:43:14.110+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Is this a loopback interface? true
┃ 2023-06-06T19:43:14.111+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Ignoring remote address on loopback int
┃ erface /fe80:0:0:0:0:0:0:1%lo0
┃ 2023-06-06T19:43:14.111+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding loopback address /0:0:0:0:0:0:0:
┃ 1%lo0
┃ 2023-06-06T19:43:14.111+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.InetAddresses] Adding loopback address /127.0.0.1
┃ 2023-06-06T19:43:14.133+0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire shared lock on daemon addr
┃ esses registry.
┃ 2023-06-06T19:43:14.134+0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
┃ 2023-06-06T19:43:14.138+0700 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
┃ 2023-06-06T19:43:14.139+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.TcpOutgoingConnector] Attempting to connect to [5ad066
┃ b6-ed42-4785-bc84-b0f5f7152b1d port:60774, addresses:[/127.0.0.1]].
┃ 2023-06-06T19:43:14.139+0700 [DEBUG] [org.gradle.internal.remote.internal.inet.TcpOutgoingConnector] Trying to connect to address /12
┃ 7.0.0.1.
┻
```

It got stuck in Trying to connect to address `127.0.0.1` for so long. :(

**Comment by TomChv** [comment]
I tried with dagger `v0.8.4`, it's still hanging :/

**Comment by wingyplus** [comment]
> I tried with dagger `v0.8.4`, it's still hanging :/

@TomChv did you try #5712?

**Comment by TomChv** [comment]
@wingyplus It fixes the issue!!!!!

`./hack/dev dagger run gradle tasks `

```
█ [1.84s] ERROR gradle tasks
┃ Starting a Gradle Daemon (subsequent builds will be faster)                                                                                                                         
┃                                                                                                                                                                                     
┃ FAILURE: Build failed with an exception.                                                                                                                                            
┃                                                                                                                                                                                     
┃ * What went wrong:                                                                                                                                                                  
┃ Directory '/Users/tomchauveau/Documents/DAGGER/dagger' does not contain a Gradle build.                                                                                             
┃                                                                                                                                                                                     
┃ A Gradle build should contain a 'settings.gradle' or 'settings.gradle.kts' file in its root directory. It may also contain a 'build.gradle' or 'build.gradle.kts' file.             
┃                                                                                                                                                                                     
┃ To create a new Gradle build in this directory run 'gradle init'                                                                                                                    
┃                                                                                                                                                                                     
┃ For more information about the 'init' task, please refer to https://docs.gradle.org/8.3/userguide/build_init_plugin.html in the Gradle documentation.                               
┃                                                                                                                                                                                     
┃ For more details on creating a Gradle build, please refer to https://docs.gradle.org/8.3/userguide/tutorial_using_tasks.html in the Gradle documentation.                           
┃                                                                                                                                                                                     
┃ * Try:                                                                                                                                                                              
┃ > Run gradle init to create a new Gradle build in this directory.                                                                                                                   
┃ > Run with --stacktrace option to get the stack trace.                                                                                                                              
┃ > Run with --info or --debug option to get more log output.                                                                                                                         
┃ > Get more help at https://help.gradle.org.                                                                                                                                         
┃                                                                                                                                                                                     
┃ BUILD FAILED in 1s                                                                                                                                                                  
┻ 
• Engine: d700473f2421 (version devel ())
• Duration: 1.845s
```

**Comment by TomChv** [comment]
@helderco @unredundant May you can double check so we can close this issue when the PR is merged

**Comment by unredundant** [comment]
<img width="684" alt="CleanShot 2023-09-01 at 08 40 13@2x" src="https://github.com/dagger/dagger/assets/5607577/ed7bc221-cff7-46a7-b4f3-a54782e29123">

still seems stuck to me :( 

on branch `iss-5666-5287`

**Comment by wingyplus** [comment]
Could you share a repository that you use in screenshot?

**Comment by wingyplus** [comment]
I have my setup here to test the pipeline in kotlin https://github.com/wingyplus/dagger-kotlin

you can run it with `./gradlew run` (support only `dagger run`.

**Comment by unredundant** [comment]
> Could you share a repository that you use in screenshot?

no repository, just ran the command inside the dagger repo (a failure is expected, since no actual gradle project).

**Comment by wingyplus** [comment]
How do you run dagger to test it? Just to be sure we’re using the same command.

**Comment by unredundant** [comment]
I ran what @TomChv ran above

```
./hack/dev dagger run gradle tasks 
```

from within the dagger repo, on the specified fix branch

**Comment by wingyplus** [comment]
I think you need what version print when running `./hack/dev dagger version`. Just need to be sure it is the devel version, not a stable one.

**Comment by gerhard** [comment]
In the last 2 weeks, we have released [`v0.8.5`](https://github.com/dagger/dagger/releases/tag/v0.8.5) and also seem to be close to finishing this:
- https://github.com/dagger/dagger/pull/5712#issuecomment-1717684840

**Comment by TomChv** [comment]
> In the last 2 weeks, we have released [v0.8.5](https://github.com/dagger/dagger/releases/tag/v0.8.5) and also seem to be close to finishing this:

https://github.com/dagger/dagger/pull/5712#issuecomment-1717684840

It seems something is wrong again, after rebased the PR from main, this change is no longer fixing the issue :/

![Screenshot 2023-09-21 at 18 56 55](https://github.com/dagger/dagger/assets/61683879/f915b17e-0d97-462d-b0ed-becab44a78da)

**Comment by TomChv** [comment]
I confirm that gradle works since #5712 has been merged:

```shell
./hack/dev dagger run gradle tasks
...
█ [4.06s] ERROR gradle tasks
┃                                                                                                                                                                    
┃ Welcome to Gradle 8.4!                                                                                                                                             
┃                                                                                                                                                                    
┃ Here are the highlights of this release:                                                                                                                           
┃  - Compiling and testing with Java 21                                                                                                                              
┃  - Faster Java compilation on Windows                                                                                                                              
┃  - Role focused dependency configurations creation                                                                                                                 
┃                                                                                                                                                                    
┃ For more details see https://docs.gradle.org/8.4/release-notes.html                                                                                                
┃                                                                                                                                                                    
┃ Starting a Gradle Daemon (subsequent builds will be faster)                                                                                                        
┃                                                                                                                                                                    
┃ FAILURE: Build failed with an exception.                                                                                                                           
┃                                                                                                                                                                    
┃ * What went wrong:                                                                                                                                                 
┃ Directory '/Users/tomchauveau/Documents/DAGGER/dagger' does not contain a Gradle build.                                                                            
┃                                                                                                                                                                    
┃ A Gradle build should contain a 'settings.gradle' or 'settings.gradle.kts' file in its root directory. It may also contain a 'build.gradle' or 'build.gradle.kts' f
┃ ile.                                                                                                                                                               
┃                                                                                                                                                                    
┃ To create a new Gradle build in this directory run 'gradle init'                                                                                                   
┃                                                                                                                                                                    
┃ For more information about the 'init' task, please refer to https://docs.gradle.org/8.4/userguide/build_init_plugin.html in the Gradle documentation.              
┃                                                                                                                                                                    
┃ For more details on creating a Gradle build, please refer to https://docs.gradle.org/8.4/userguide/tutorial_using_tasks.html in the Gradle documentation.          
┃                                                                                                                                                                    
┃ * Try:                                                                                                                                                             
┃ > Run gradle init to create a new Gradle build in this directory.                                                                                                  
┃ > Run with --stacktrace option to get the stack trace.                                                                                                             
┃ > Run with --info or --debug option to get more log output.                                                                                                        
┃ > Get more help at https://help.gradle.org.                                                                                                                        
┃                                                                                                                                                                    
┃ BUILD FAILED in 1s                                                                                                                                                 
┻ 
• Engine: 5096d9fac428 (version devel ())

```
